zeroconf/_handlers.py

```diff
@@ -105,30 +106,45 @@ def _answer_non_ptr_query(self, msg: DNSIncoming, out: DNSOutgoing, question: DN
             for dns_address in service.dns_addresses():
                 out.add_additional_answer(dns_address)
 
-    def response(self, msg: DNSIncoming, unicast: bool) -> Optional[DNSOutgoing]:
+    def response(
+        self, msg: DNSIncoming, addr: Optional[str], port: int
+    ) -> Tuple[Optional[DNSOutgoing], Optional[DNSOutgoing]]:
         """Deal with incoming query packets. Provides a response if possible."""
-        if unicast:
```

Comment:
Now that its separated into unicast and multicast we can fix https://github.com/jstasiak/python-zeroconf/issues/500

The following need to be fixed first though:
https://github.com/jstasiak/python-zeroconf/issues/595
https://github.com/jstasiak/python-zeroconf/issues/464